### PR TITLE
feat(connections): add GitHub App auth for GitHub Enterprise (bots/agents)

### LINKED ISSUE #2978: Make GitHub Enterprise connector available for bots/agents
## Context

A GitHub **Enterprise** connector already exists in DAM, **but it is not available for bots/agents** — only for interactive/user auth. As a result, the `dam-application[bot]` identity authenticates only against `github.com` and cannot reach GHE-hosted repositories, so it can't read or open PRs/issues on repos that live on a GitHub Enterprise instance.

## Ask

Make the existing GitHub Enterprise connector usable by **bots/agents** (not just interactive users), so DAM can read + open PRs/issues on GHE-hosted repos.

## Notes / possible scope
- Support app-install / bot credentials against the enterprise host (not only user OAuth).
- Configurable enterprise host / API base URL (GHE Server or Enterprise Cloud).
- Reuse the existing GHE connector; extend the auth path for the bot identity.

_Filed from the Slack thread on request of the team._

### PR DESCRIPTION
**Pull request by pilartomas**
## Summary

- Closes #2978. The GitHub Enterprise connector only supported interactive OAuth, so bot/scheduled-agent identities (e.g. automated PR/issue creation) couldn't authenticate against a GHE instance — only against github.com.
- Adds a `github-enterprise-app` connection template: the same GitHub App installation auth (app id + installation id + private key, no user consent step) already offered for github.com via `github-app`, now with the enterprise host taken as user input. The platform mints installation tokens (`ghs_…`) from the app's private key exactly as it already does for github.com.
- The host is substituted through `apiBaseUrl` and the two GHE host injections (`api.{host}` Bearer for REST, apex `{host}` Basic x-access-token for git-over-HTTPS), mirroring the existing interactive `github-enterprise` OAuth path's host handling — factored into a shared `githubEnterpriseHostContributions` helper so both auth modes stay in sync.
- Wires the new `host` input through the connection-template input list, the API create-input schema, the UI create form, and the CLI (`dam connection connect github-enterprise-app --host ghe.acme.com ...`).
- The operator-preset GHE host (`oauthAppDefaults.githubEnterprise.host`, already wired for the OAuth template) is reused as a preset for this template too, so operators don't configure the enterprise host twice.
- Gated behind the advanced-connections experimental flag, same as the existing `github-app` (github.com) template — this is a UI-catalog-offering gate only, not a backend restriction.

## Test plan

- [x] `packages/api-server`: new unit test file (`connections-github-enterprise-app-template.test.ts`) covering host substitution into `apiBaseUrl`/`auth.host`, the GH_TOKEN/GH_HOST + host-injection contributions, missing-host rejection, operator-preset host fallback, and the template-view input states (required vs. overridable). Full suite: 629 passed (2 pre-existing, unrelated `cluster-ca-probe` failures reproduce on `main`).
- [x] `packages/api-server-api`, `packages/ui`, `packages/cli`: `tsc --noEmit`, `eslint`, `prettier --check` all clean on touched files.
- [x] `packages/ui` (104 tests) and `packages/cli` (154 tests) full suites pass with no regressions.
- [ ] Manual: create a `github-enterprise-app` connection against a real GHE instance and confirm an agent can read/open a PR on it (not done — no live GHE instance available in this environment).

### CODE REVIEW

File: packages/ui/src/modules/connections/lib/build-create-payload.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export function buildCreatePayload(
       return compact({
         ...common,
         authKind: "github-app" as const,
```
Comment (dam-code-guardian) [inline packages/ui/src/modules/connections/lib/build-create-payload.ts:77]: 🟡 **Warning:** This branch returns friendly inline errors for empty `appId`/`installationId`/`privateKey` (lines above), but `host` is passed straight through `submitted("host")` — when the `host` input is `required` (GHE, no operator preset) and left empty, `compact()` drops it and the server rejects with a raw thrown `template github-enterprise-app: missing host` string. The server is safe, but the user sees a server error rather than an inline "Host is required". Consider gating it like the other three:

```suggestion
        authKind: "github-app" as const,
        host: submitted("host"),
```

(or add `if (inputsByName.get("host")?.state === "required" && !submitted("host")) return { error: "Host is required" };` before the return).

**Review by dam-code-guardian — COMMENTED** [review]
🛡️ **Code Guardian 🛡️** — ⚠️ Code Review @ `a60cf1f`

## PR #2984: feat(connections): add GitHub App auth for GitHub Enterprise (bots/agents)
**Author:** pilartomas | **Branch:** feat/ghe-app-bot-auth → main | **Changes:** +264 −21 (8 files)

### Summary
Adds a `github-enterprise-app` connection template so bot/scheduled-agent identities can authenticate against a GHE instance via GitHub App installation tokens (app id + installation id + private key), with the enterprise host taken as user input. The host is substituted through `apiBaseUrl` and the two GHE host injections, factored into a shared `githubEnterpriseHostContributions` helper so the App and OAuth GHE modes stay in sync.

### Findings
- ✅ **Looks good:** The host-substitution logic in `buildGitHubApp` is correct and safely gated: `needsHost` keys off `{host}` in `apiBaseUrl`, so the fixed github.com `github-app` template (no placeholder) is untouched and keeps its 4 static injections including `raw.githubusercontent.com`, while the enterprise template gets `api.{host}` Bearer + apex Basic. Missing host throws server-side (`build-connection.ts:463`), verified by the "rejects a missing host" test.
- ✅ **Looks good:** `githubEnterpriseHostContributions` extraction genuinely de-dupes the OAuth GHE path and the new App path — the two auth modes now share one source of truth for GH_HOST + the two host injections.
- ✅ **Looks good:** Type safety holds — `host?: string` added as `.min(1).optional()` on the shared `githubAppCreateInput`, threaded through the discriminated union cleanly; no `any` introduced (the lone `as` is in test JSON parsing). Private key still flows to a `SecretRef` and is never logged.
- 🟡 **Warning:** UX validation asymmetry in the create form (`packages/ui/src/modules/connections/lib/build-create-payload.ts:73`). The `github-app` branch returns friendly inline errors for empty `appId`/`installationId`/`privateKey`, but an empty **required** `host` is silently dropped by `compact()` and reaches the server, which rejects it with a raw thrown `template github-enterprise-app: missing host`. The server is safe (no security impact), but for a `required` host the user gets a server error string instead of an inline "Host is required". Consider gating `host` the same way as the other three fields.
- 🟢 **Suggestion:** `buildGitHubApp` uses `throw new Error(...)` for validation while the architecture favors `Result` types in the domain layer — this matches the pre-existing style in the same function (appId/installationId/privateKey already throw), so it's consistent, not a regression; noting only for a future domain-error sweep.

### TypeScript Engineering Review
✅ No architecture-rule violations introduced. Validation stays in the contract package (`schemas.ts` Zod), host-substitution logic stays in the pure domain layer (`build-connection.ts`, `catalog.ts`, `connection-template.ts`), and the CLI/UI payload builders map validated input without business logic. The `githubEnterpriseHostContributions` helper is a clean DRY consolidation. One 🟡 (UX validation asymmetry) and one 🟢 (throw vs `Result`, pre-existing style) noted above.

### Verdict
COMMENT — solid, well-tested, security-conscious feature that faithfully mirrors the existing OAuth GHE path; the one 🟡 is a create-form UX nit (empty required host → server error instead of inline), not a correctness or security issue.

---
_Review by [Code Guardian 🛡️](https://github.com/dam-agents/code-guardian) · automated code guardian_

File: packages/ui/src/modules/connections/lib/build-create-payload.ts (hunk 1 above)
Comment (pilartomas) [reply packages/ui/src/modules/connections/lib/build-create-payload.ts:77]: Fixed in 613106c5 — an empty required `host` now returns the same inline "Host is required" error the other fields already get, instead of reaching the server's raw throw. Also added unit test coverage for `buildCreatePayload` (none existed before), which needed a small `vitest.config.ts` fix (`@` alias was missing, only defined in `vite.config.ts`/`tsconfig.json`).

**Review by dam-code-guardian — APPROVED** [review]
🛡️ **Code Guardian 🛡️** — ✅ Code Review @ `613106c`

## PR #2984: feat(connections): add GitHub App auth for GitHub Enterprise (bots/agents)
**Author:** pilartomas | **Branch:** feat/ghe-app-bot-auth → main

### Summary
Adds a `github-enterprise-app` connection template so bots/scheduled agents can auth against GHE via GitHub App installation tokens, host substituted through `apiBaseUrl` + the two GHE host injections via a shared `githubEnterpriseHostContributions` helper.

### Changes since last review
Previous HEAD: `a60cf1f` (2026-07-27T11:02:01Z) — verdict COMMENT

- ✅ **Fixed:** Empty required `host` now returns the inline `"Host is required"` error before payload build, matching the sibling `appId`/`installationId`/`privateKey` checks — no longer falls through to the server's raw throw (`build-create-payload.ts:69`). The guard correctly keys on `host` being `required`, so the operator-preset (`overridable`) case is untouched; covered by the new `build-create-payload.test.ts` (4 cases incl. preset + passthrough).
- 🔁 **Still present:** `buildGitHubApp` uses `throw` rather than a `Result` type — 🟢 only, consistent with the pre-existing style in that function; future domain-error-sweep note (`build-connection.ts`).

### Findings
_No new findings at this HEAD._

### Documentation Check (doc-drift)
✅ Aligned — no docs touched; the new catalog template composes the already-documented GitHub App auth + GHE-host mechanisms, no subsystem behavior/responsibility change.

### TypeScript Engineering Review
✅ No new architecture violations. The added `host` validation stays in the UI payload-builder layer and mirrors the existing discriminated-union input checks; the `@`-alias added to `vitest.config.ts` mirrors the existing vite/tsconfig alias (test infra). 🔁 1 finding from the previous review still present (🟢 throw-vs-`Result`, see review at `a60cf1f`).

### Verdict
APPROVE — the prior 🟡 create-form UX asymmetry is fixed with matching inline validation and unit-test coverage; only a pre-existing 🟢 style note remains.

---
_Review by [Code Guardian 🛡️](https://github.com/dam-agents/code-guardian) · automated code guardian_
